apps/condo/domains/notification/utils/serverSchema/index.js

```diff
@@ -151,13 +153,44 @@ async function checkMessageTypeInBlackList (context, message) {
         if (isMessageTypeInUserBlackList) return { error: MESSAGE_TYPE_IN_USER_BLACK_LIST }
     }
 
+    //TODO(Kekmus) think more about it maybe in some cases it doesnt work
+    if (message.meta.data && message.meta.data.B2CAppId) {
+        const messageAppBlackListWhere = {
+            type: message.type,
+            OR: [
+                { app_is_null: true },
+                { app: message.meta.data.B2CAppId },
+            ],
+            deletedAt: null,
+        }
+
+        const messageAppBlackListRules = await MessageAppBlackList.getAll(context, messageAppBlackListWhere)
```

Comment:
It's better to use `count` to prevent retrieving all fields.
```suggestion
        const messageAppBlackListRules = await MessageAppBlackList.count(context, messageAppBlackListWhere)
```